app/src/main/res/layout/activity_main.xml

```diff
@@ -47,30 +47,44 @@
         <TextView
             android:id="@+id/textView"
             android:layout_width="0dp"
-            android:layout_height="@dimen/camera_view_height"
+            android:layout_height="0dp"
             android:layout_marginStart="@dimen/camera_view_height"
             android:gravity="center_vertical"
             android:textAlignment="gravity"
-            android:textSize="24sp"
+            android:autoSizeTextType="uniform"
```

Comment:
Hello! I understand what you mean. And I guess we can change this code I added before:
```
android:autoSizeTextType="uniform"
```
to:
```
app:autoSizeTextType="uniform"
```
and change `TextView` to `androidx.appcompat.widget.AppCompatTextView`. In the meantime, add a new implementation in the gradle.build file inside the app folder:
```
implementation 'androidx.appcompat:appcompat:1.3.1'
```
In this way, I believe it won't affect the compatibility of the earlier version of Android. Here is the screenshot after this modification on the Pixel 2 with Android 7:

![Screenshot_1630136592](https://user-images.githubusercontent.com/25502419/131210634-41b3d64a-7b00-4d3d-bae7-a28e5383f27b.png)
![Screenshot_1630136603](https://user-images.githubusercontent.com/25502419/131210635-0bfe89a9-aed6-46b6-aa64-2238c6e9c2e6.png)
![Screenshot_1630136605](https://user-images.githubusercontent.com/25502419/131210636-6a067c1c-c682-419c-acac-ca34a861e941.png)

Maybe it could solve the problem you mentioned before. I would like to know how do you think about this. Thank you so much!! :)

Reply:
This is an acceptable resolution too. The only reason I suggested 24dp again was to match the UI title size for 'Pulse:' text, if you use AutoSize it is much smaller than the text below.

I'm fine with it either way, I only suggest 24dp because this was what the original author had before. UI is subjective so up to them. I'm good with either implementation.